Adding how to set an error with opentracing and java (#5894)
DataDog/documentation: content/en/tracing/advanced/opentracing.md

  }
}
```

Notice the above examples only use the OpenTracing classes. Check the [OpenTracing API][1] for more details and information.

#### Set errors

> To customize an error associated to one of your span, you must import the `io.opentracing.Span`, `io.opentracing.tag.Tags`, and `io.opentracing.util.GlobalTracer` library in the method where the error occurs:

Reviewer: ```suggestion
To customize an error associated to one of your spans, import the `io.opentracing.Span`, `io.opentracing.tag.Tags`, and `io.opentracing.util.GlobalTracer` libraries into the method where the error occurs:
```

```java
import io.opentracing.Span;
import io.opentracing.tag.Tags;
import io.opentracing.util.GlobalTracer;
```

> Then you need to set the error to `true` and add *at least* the `error.msg`, `error.type`, and `error.stack` tag to your span.

Reviewer: ```suggestion
Then, set the error to `true` and add *at least* the `error.msg`, `error.type`, and `error.stack` tags to your span.
```

```java
    final Span span = GlobalTracer.get().activeSpan();
    if (span != null) {
      Tags.ERROR.set(span, true);
      span.log(Collections.singletonMap(Fields.ERROR_OBJECT, ex));
    }
```

> **Note**: Any relevant error metadata explained in [the Trace View docs][4] can also be added.

Reviewer: ```suggestion
**Note**: Any relevant error metadata explained in the [Trace View docs][4] can also be added.
```

> If the current span isn't the root span, mark it as an error by using the `dd-trace-api` library to grab the root span with `MutableSpan` and use `setError(true)` (See [source code][5]).

Reviewer: ```suggestion
If the current span isn't the root span, mark it as an error by using the `dd-trace-api` library to grab the root span with `MutableSpan`, then use `setError(true)`. See the [source code][5] for more details.
```

  }
}
```

Notice the above examples only use the OpenTracing classes. Check the [OpenTracing API][1] for more details and information.

#### Set errors

To customize an error associated to one of your span, you must import the `io.opentracing.Span`, `io.opentracing.tag.Tags`, and `io.opentracing.util.GlobalTracer` library in the method where the error occurs:

```java
import io.opentracing.Span;
import io.opentracing.tag.Tags;
import io.opentracing.util.GlobalTracer;
```

Then you need to set the error to `true` and add *at least* the `error.msg`, `error.type`, and `error.stack` tag to your span.

```java
    final Span span = GlobalTracer.get().activeSpan();
    if (span != null) {
      Tags.ERROR.set(span, true);
      span.setTag("error.msg", "<ERROR_MESSAGE>");
      span.setTag("error.type", "<EXCEPTION_NAME>");

>       span.setTag("error.stack", "<ERROR_STACK>");

Reviewer 2: Replace these with
```java
span.log(Collections.singletonMap(Fields.ERROR_OBJECT, ex));
```

to better align with OpenTracing: https://github.com/opentracing/opentracing-java
